agent/wsclient/client_test.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2014-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package wsclient
+
+import (
+	"io"
+	"os"
+	"testing"
+
+	"github.com/aws/amazon-ecs-agent/agent/acs/model/ecsacs"
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/aws/amazon-ecs-agent/agent/wsclient/mock/utils"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+
+	"github.com/gorilla/websocket"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// Concurrent writes will cause a panic in the websocket library
+// See https://godoc.org/github.com/gorilla/websocket#hdr-Concurrency
+func TestConcurrentWritesDontPanic(t *testing.T) {
+	mockServer, _, requests, _, _ := mockwsutils.StartMockServer(t, make(chan []byte))
+	defer mockServer.Close()
+
+	req := ecsacs.AckRequest{Cluster: aws.String("test"), ContainerInstance: aws.String("test"), MessageId: aws.String("test")}
+
+	cs := getClientServer(mockServer.URL)
+	cs.Connect()
+
+	executeTenRequests := func() {
+		for i := 0; i < 10; i++ {
+			cs.MakeRequest(&req)
+		}
+	}
+
+	go executeTenRequests()
```

Comment:
Ah! My goal was to force a concurrent write. I was thinking that the easiest way to achieve this was by having one group of writes in a routine and the other group in the main thread. If it makes the intent more clear, I can put them both in goroutines.